core/src/main/java/org/mule/api/registry/ServiceRegistry.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.registry;
+
+import java.util.Collection;
+import java.util.ServiceLoader;
+
+/**
+ * API component to be used as standard SPI discovery mechanism.
+ * Locates and returns all registered instances of a given provider
+ * class. Advantages of using this interface instead of hitting something
+ * like {@link ServiceLoader} directly are the ability to inject a different
+ * lookup mechanism and to facilitate testing.
+ * <p/>
+ * Notice that in future versions of Mule this interface might me moved to
```

Comment:
This paragraph should be removed because we don't know if it will be moved, and if it is, the there will be a deprecation/migration guide as needed